controllers/vaultdynamicsecret_controller.go

```diff
@@ -221,11 +221,23 @@ func (r *VaultDynamicSecretReconciler) isRenewableLease(resp *secretsv1alpha1.Va
 	return true
 }
 
-func (r *VaultDynamicSecretReconciler) syncSecret(ctx context.Context, vClient vault.Client, o *secretsv1alpha1.VaultDynamicSecret) (*secretsv1alpha1.VaultSecretLease, error) {
+func (r *VaultDynamicSecretReconciler) syncSecret(
+	ctx context.Context, c vault.ClientBase, o *secretsv1alpha1.VaultDynamicSecret,
+) (*secretsv1alpha1.VaultSecretLease, error) {
 	path := fmt.Sprintf("%s/%s", o.Spec.Mount, o.Spec.Path)
-	resp, err := vClient.Read(ctx, path)
-	if err != nil {
-		return nil, err
+	var err error
+	var resp *api.Secret
+	if len(o.Spec.Params) > 0 {
```

Comment:
I like that params implies POST, as that will feel familiar from tools like curl. However, not all POST endpoints require parameters, e.g. AWS STS creds can be a POST with no params: https://developer.hashicorp.com/vault/api-docs/secret/aws#generate-credentials

And the Kubernetes secrets engine will also soon allow generating creds without any params: https://github.com/hashicorp/vault-plugin-secrets-kubernetes/pull/27

Reply:
Woah, that could make things tricky. I wonder if we want to add an option to make the method explicit.

Reply:
Yeah that was my thought too, so people get the current behaviour as the default but they can also override the method.